### PR TITLE
Ana/add missing security headers

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Following the mozilla scan advices. [Here ](https://observatory.mozilla.org/analyze/app.lunie.io)is the current report.

I'm only unsure about the `Content-Security-Policy` header. This one has a lot of possible combinations that improve a lot security. But I'm uncertain if we should include 'unsafe-inline' for example... In local everything is working fine setting everything to `self` it seems.

Here are the `Content-Security-Policy` header [docs](https://infosec.mozilla.org/guidelines/web_security#content-security-policy). I followed the fifth example

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by Bitcoinera** [comment]
[More docs](https://content-security-policy.com/) on the `Content-Security-Policy` header (it is a very complex one)

File: public/_headers [hunk 1]
```diff
@@ -1,7 +1,9 @@
 /*
   X-Frame-Options: DENY
   X-XSS-Protection: 1; mode=block
-  Content-Security-Policy: frame-ancestors 'self'
+  Content-Security-Policy: frame-ancestors 'self'; default-src 'self'; script-src 'self'
```
Comment (faboweb) [inline public/_headers:4]: did you consider https://github.com/luniehq/lunie/blob/develop/vue.config.js#L55 ?

Comment (Bitcoinera) [reply public/_headers:4]: I see, it is really well configured there... But checking app.lunie.io in Network I only get the header present in `headers`:



Also, I don't understand, why don't we put all those options directly in `headers`? Why do we need to do it through webpack?

Maybe it is not working

Comment (faboweb) [reply public/_headers:4]: let'S move it all over to the netlify config then

**Comment by faboweb** [comment]
the file is invalid now according to Netlify

**Comment by Bitcoinera** [comment]
> the file is invalid now according to Netlify

Fixed. This playground was useful: https://play.netlify.com/headers

**Comment by Bitcoinera** [comment]
> doesn't work

Oops, should have checked earlier. Will fix it tomorrow 👌

File: public/_headers [hunk 2]
```diff
@@ -5,7 +5,7 @@
   Content-Security-Policy: object-src 'none'; 
   Content-Security-Policy: base-uri 'self'; 
   Content-Security-Policy: default-src 'self; 
-  Content-Security-Policy: script-src 'self' https://*.lunie.io; 
+  Content-Security-Policy: script-src 'self' https://*.lunie.io https://deploy-preview-*--lunieio.netlify.app;
```
Comment (faboweb) [inline public/_headers]:
```suggestion
  Content-Security-Policy: script-src 'self' https://*.lunie.io https://*--lunieio.netlify.app; 
```

Comment (Bitcoinera) [reply public/_headers]: yeah, that wasn't working 👍 

Now that seems fixed, but the problem is that I'm getting a `Note that 'img-src' was not explicitly set, so 'default-src' is used as a fallback` when the `img-src` is already set and, in fact, even before the actual `default-src`.

Really weird.

Will continue looking into this tomorrow. Cannot explain why this happens

File: public/_headers [hunk 3]
```diff
@@ -4,9 +4,9 @@
   Content-Security-Policy: frame-ancestors 'self';
   Content-Security-Policy: object-src 'none'; 
   Content-Security-Policy: base-uri 'self'; 
-  Content-Security-Policy: img-src * 'self' https://www.google-analytics.com;
+  Content-Security-Policy: img-src 'self' https://www.google-analytics.com data:;
   Content-Security-Policy: manifest-src 'self' https://*.lunie.io https://*.netlify.app;
-  Content-Security-Policy: script-src-elem * 'unsafe-inline';
+  Content-Security-Policy: script-src-elem 'unsafe-inline' data:;
```
Comment (faboweb) [inline public/_headers:9]: `data:` is considered dangerous I think lets investigate and maybe change

Comment (Bitcoinera) [reply public/_headers:9]: not so easy to set these headers. That is for sure

**Comment by faboweb** [comment]
still doesn't work

**Comment by Bitcoinera** [comment]
> still doesn't work

I know xD I was fixing Polkadot bugs

**Comment by faboweb** [comment]
validator images are not loading

**Comment by Bitcoinera** [comment]
> validator images are not loading

Yes, and loading is somehow so slow... CSP are tricky, but also powerful. Maximum security 🔐

**Comment by Bitcoinera** [comment]
Now this is the only thing "failing"

image

**Comment by Bitcoinera** [comment]
Can you also check the load speed? Seems to have decreased on first run... or maybe this is how deploys are anyways?

**Comment by faboweb** [comment]
> Now this is the only thing "failing"

that is a different error

**Comment by faboweb** [comment]
image

**Comment by jbibla** [comment]
thank you for your initiative here @Bitcoinera

**Comment by Bitcoinera** [comment]
> thank you for your initiative here @Bitcoinera

🙏 

This is actually ready to be merged! Let's do it! 💃

**Comment by faboweb** [comment]
image

**Comment by Bitcoinera** [comment]
I cannot quote reply, but here you have the explanation to that mysterious error: it comes from the Grammarly extension: https://stackoverflow.com/questions/45366744/refused-to-load-the-font-datafont-woff-it-violates-the-following-content

**Comment by Bitcoinera** [comment]
I cannot find any other errors. Intercom I guess we can only test on production right? But should be fine

**Comment by faboweb** [comment]
But we need to work away from the data attribute: https://security.stackexchange.com/questions/94993/is-including-the-data-scheme-in-your-content-security-policy-safe/95011#95011

**Pull request merged**